caffeine/src/javaPoet/java/com/github/benmanes/caffeine/cache/NodeFactoryGenerator.java

```diff
@@ -206,7 +206,7 @@ private static Set<List<Object>> combinations() {
   private static String encode(String className) {
     return Feature.makeEnumName(className)
         .replaceFirst("STRONG_KEYS", "P") // puissant
-        .replaceFirst("WEAK_KEYS", "F") // faible
+        .replaceFirst("WEAK_KEYS", "F") // failable
```

Comment:
Note that the normal word is `fallible`, but I'm not really sure what this project is doing and it was already using `failible` ...

Reply:
The contributor who refactored this code (https://github.com/ben-manes/caffeine/pull/200) used french words to avoid name collisions. This context is for a weak reference which allows the garbage collector to discard it when there are no strong references.

> "Faible" is a French word meaning weak, and "failable" is an English word meaning liable to make mistakes or fail. The key difference is that "faible" describes weakness, while "failable" describes the potential to fail.

Reply:
sgtm, I'd probably use `persistent` since this is about strong-weak-soft references for whether the instance can disappear underneath you.